addons/sale/models/sale_order_line.py

```diff
@@ -944,15 +944,28 @@ def _compute_qty_to_invoice(self):
         """
         Compute the quantity to invoice. If the invoice policy is order, the quantity to invoice is
         calculated from the ordered quantity. Otherwise, the quantity delivered is used.
+        For combo product lines, first compute all other lines, and then set quantity to invoice
+        only if at least one of its combo item lines is invoiceable.
         """
+        combo_lines = []
         for line in self:
             if line.state == 'sale' and not line.display_type:
-                if line.product_id.invoice_policy == 'order':
+                if line.product_id.type == 'combo':
+                    combo_lines.append(line)
+                elif line.product_id.invoice_policy == 'order':
                     line.qty_to_invoice = line.product_uom_qty - line.qty_invoiced
                 else:
                     line.qty_to_invoice = line.qty_delivered - line.qty_invoiced
             else:
                 line.qty_to_invoice = 0
+        for combo_line in combo_lines:
+            if any(
+                line.combo_item_id and line.qty_to_invoice
+                for line in combo_line.linked_line_ids
+            ):
+                combo_line.qty_to_invoice = combo_line.product_uom_qty - combo_line.qty_invoiced
```

Comment:
This is not working :( When combo product with all `delivery` invoice_policy combo added in SO
Because compute for combo line only called when creating new SO or changing state as we don't change other dependent field for combo lines but it should called for combo line when we are changing qty_delivered on it's related combo lines :thinking: